clang/lib/Sema/SemaConcept.cpp

```diff
@@ -585,8 +585,8 @@ static bool CheckConstraintSatisfaction(
 
   ArrayRef<TemplateArgument> TemplateArgs =
       TemplateArgsLists.getNumSubstitutedLevels() > 0
-          ? TemplateArgsLists.getOutermost()
-          : ArrayRef<TemplateArgument> {};
+          ? TemplateArgsLists.getInnermost()
```

Comment:
This looks strange, can you explain what is going on here?

Reply:
The innermost template arguments are the ones that actually pertain to the template we are instantiating... the outermost arguments pertain to some unrelated containing template.

Reply:
Right, sure... I guess I'm confused as to the old implementation then.  Was it just wrong?

Reply:
Right, sure... I guess I'm confused as to the old implementation then.  Was it just wrong?

Reply:
Yeah, the old implementation was just wrong.

Reply:
Ok, makes sense.  I am SOMEWHAT remembering that at the time we assumed that >0 meant ==1 at the time, and 'getInnermost' wasn't actually in existence at the time.  So either it was typed confusingly in the 1st place, or things have changed in the meantime.  Either way, probably  bitrot here.